azure-proto-core-test/UserAssignedIdentityTests.cs

```diff
@@ -0,0 +1,377 @@
+﻿using azure_proto_core.Resources;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace azure_proto_core_test
+{
+    public class UserAssignedIdentityTests : UserAssignedIdentity
```

Comment:
Per my comment on SystemAssignedIdentities, you should avoid having this class extend the class it is testing.  If you need a custom implementation of that class, consider having a test artifact used by this class.